lib/linguist/heuristics.yml

```
@@ -163,6 +171,11 @@ disambiguations:
   - language: Frege
     pattern: '^\s*(import|module|package|data|type) '
   - language: Text
+- extensions: ['.frm']
+  rules:
+  - language: FreeBasic
+    pattern: '^\s*#(define|endif|endmacro|if|ifdef|ifndef|include|lang|macro)\s'
+  - language: VBA
```

<a name="yup-this-reason"></a>This should be removed:

```suggestion
- - language: VBA
```

A `language` field without an accompanying `pattern` matches *any* file that doesn't match a preceding pattern. That is to say, it behaves like this:

~~~python
if pattern.match(candidateFile)
	return FreeBasic
else
	return VBA
~~~

It's okay if none of the `rules` match a candidate, so long as our classifier does a good job of analysing samples. 😉

---

The test was giving error, so I added these lines
- language: VBA
- language: C++